src/Services/PatientService.php

```diff
@@ -173,6 +175,77 @@ public function databaseInsert($data)
         }
     }
 
+    /**
+     * Grants portal access to patients in case they do
+     * not have the permission.
+     *
+     * @param $puuidString
+     * @return void
+     */
+    public function allowPatientPortal($puuidString)
+    {
+        $sql = "UPDATE `patient_data` SET";
+        $sql .= " `allow_patient_portal`='YES'";
+        $sql .= " WHERE `uuid` = ?";
+        $puuidBinary = UuidRegistry::uuidToBytes($puuidString);
+        sqlQuery($sql, array($puuidBinary));
+    }
+
+    /**
+     * Verifies if the patient has access to the portal.
+     *
+     * @param $patient_data
+     * @return boolean
+     */
+    public function hasPortalPermission($patient_data)
+    {
+        $patient_portal_permission = strtolower($patient_data["allow_patient_portal"]);
+        $has_portal_permission = "yes";
```

Comment:
Use a constan for this string "yes". 
const string private HAS_PORTAL_PERMISSION = "yes";
Then you can use on the class:
$has_portal_permission = self::HAS_PORTAL_PERMISSION;